qiita_db/support_files/patches/26.sql

```diff
@@ -0,0 +1,25 @@
+-- June 11,, 2015
+-- Adds portal type to analyses
+
+-- Temporarily have default value for non-portaled analyses
+ALTER TABLE qiita.analysis
+ADD portal_type_id bigint NOT NULL DEFAULT 1;
+
+-- Add FK to portal_type_id
+ALTER TABLE qiita.analysis
+ADD CONSTRAINT fk_analysis_portal
+FOREIGN KEY (portal_type_id)
+REFERENCES qiita.portal_type(portal_type_id);
+
+-- Drop defaut value constraint so it is now required to give portal type
+ALTER TABLE qiita.analysis
+ALTER portal_type_id DROP DEFAULT;
+
+-- Remove existing portals and replace with more relevant ones
+UPDATE qiita.study SET portal_type_id = 1 WHERE portal_type_id = 3;
```

Comment:
It’s critical that the same study can be seen in multiple portals.

> On Jun 16, 2015, at 9:09 AM, Antonio Gonzalez <notifications@github.com> wrote:
> 
> In qiita_db/support_files/patches/26.sql <https://github.com/biocore/qiita/pull/1256#discussion_r32537526>:
> 
> > +-- Temporarily have default value for non-portaled analyses
> > +ALTER TABLE qiita.analysis
> > +ADD portal_type_id bigint NOT NULL DEFAULT 1;
> > +
> > +-- Add FK to portal_type_id
> > +ALTER TABLE qiita.analysis
> > +ADD CONSTRAINT fk_analysis_portal
> > +FOREIGN KEY (portal_type_id)
> > +REFERENCES qiita.portal_type(portal_type_id);
> > +
> > +-- Drop defaut value constraint so it is now required to give portal type
> > +ALTER TABLE qiita.analysis
> > +ALTER portal_type_id DROP DEFAULT;
> > +
> > +-- Remove existing portals and replace with more relevant ones
> > +UPDATE qiita.study SET portal_type_id = 1 WHERE portal_type_id = 3;
> Also, this will mean that a study can only belong to one portal, which I'm not sure is what we need/want. I can see having 4 portals: qiita (all studies), emp, sloan, hmp2 and having studies that belong to the 3 of the 4 (qiita, sloan, hmp2).
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/biocore/qiita/pull/1256/files#r32537526>.
>